pkgs/ffigen/lib/src/config_provider/spec_utils.dart

```diff
@@ -30,15 +30,15 @@ String _replaceSeparators(String path) {
   }
 }
 
-/// Replaces the path separators according to current platform. If a relative
-/// path is passed in, it is resolved relative to the config path, and the
-/// absolute path is returned.
-String _normalizePath(String path, String? configFilename) {
-  final skipNormalization =
+/// Replaces the path separators according to current platform, and normalizes .
+/// and .. in the path. If a relative path is passed in, it is resolved relative
+/// to the config path, and the absolute path is returned.
+String normalizePath(String path, String? configFilename) {
+  final resolveInConfigDir =
       (configFilename == null) || p.isAbsolute(path) || path.startsWith('**');
-  return _replaceSeparators(skipNormalization
+  return _replaceSeparators(p.normalize(resolveInConfigDir
       ? path
-      : p.absolute(p.join(p.dirname(configFilename), path)));
+      : p.absolute(p.join(p.dirname(configFilename), path))));
```

Comment:
off topic: `p.join` doesn't normalize, that's why `resolve` on `Uri` is more powerful and a better default to use than `join` imho. But since this code is already using `package:path` no need to change it now.

Maybe do a quick search over the codebase to see where `join` is called and if there's any other places which don't deal well with relative paths.

Reply:
Done